blockdevice/stats.go

```diff
@@ -474,3 +475,9 @@ func (fs FS) SysBlockDeviceUnderlyingDevices(device string) (UnderlyingDeviceInf
 	return UnderlyingDeviceInfo{DeviceNames: underlying}, nil
 
 }
+
+// SysBlockDeviceSize returns the size of the block device from /sys/block/<device>/size.
+// Note that this needs to be multiplied by LogicalBlockSize to get the size in bytes.
```

Comment:
This size is [reported](https://elixir.bootlin.com/linux/v6.9.8/source/block/genhd.c#L944) in units of `sector_t`, which are 512 bytes.
```c
ssize_t part_size_show(struct device *dev,
		       struct device_attribute *attr, char *buf)
{
	return sprintf(buf, "%llu\n", bdev_nr_sectors(dev_to_bdev(dev)));
}
```
`sector_t` is [defined](https://elixir.bootlin.com/linux/v6.9.8/source/include/linux/types.h#L134) as:
```c
/**
 * The type used for indexing onto a disc or disc partition.
 *
 * Linux always considers sectors to be 512 bytes long independently
 * of the devices real block size.
 *
 * blkcnt_t is the type of the inode's block count.
 */
typedef u64 sector_t;
...
```
See [also](https://elixir.bootlin.com/linux/v6.9.8/source/include/linux/blk_types.h#L24):
```c
/*
 * The basic unit of block I/O is a sector. It is used in a number of contexts
 * in Linux (blk, bio, genhd). The size of one sector is 512 = 2**9
 * bytes. Variables of type sector_t represent an offset or size that is a
 * multiple of 512 bytes. Hence these two constants.
 */
#ifndef SECTOR_SHIFT
#define SECTOR_SHIFT 9
#endif
#ifndef SECTOR_SIZE
#define SECTOR_SIZE (1 << SECTOR_SHIFT)
#endif
```

Reply:
So it is possible for the logical block size to be reported as e.g. 4096 for an Advanced Format (AF) disk, but the `/sys/block/<dev>/size` will still always be reported in units of 512-byte sectors.